src/OgAccess.php

```diff
@@ -0,0 +1,278 @@
+   *   with "administer group" will be granted all permissions.
+   *   Defaults to FALSE.
+   *
+   * @return \Drupal\Core\Access\AccessResultInterface
+   *   TRUE or FALSE if the current user has the requested permission.
+   *   NULL, if the given group isn't a valid group.
+   */
+  public static function userAccess(EntityInterface $group_entity, $operation, AccountInterface $user = NULL, $skip_alter = FALSE, $ignore_admin = FALSE) {
+    $group_type_id = $group_entity->getEntityTypeId();
+    $bundle = $group_entity->bundle();
+
+    if (!Og::isGroup($group_type_id, $bundle)) {
+      // Not a group.
+      return AccessResult::neutral()->addCacheableDependency($group_entity);
+    }
+
+    $user = $user ?: \Drupal::currentUser()->getAccount();
+    $user_id = $user->id();
+
+    // User ID 1 has all privileges.
+    if ($user_id == 1) {
+      return AccessResult::allowed();
+    }
+
+    // Administer group permission.
+    if (!$ignore_admin) {
+      $user_access = AccessResult::allowedIfHasPermission($user, static::ADMINISTER_GROUP_PERMISSION);
+      if ($user_access->isAllowed()) {
+        return $user_access;
+      }
+    }
+
+    // Group manager has all privileges (if variable is TRUE) and they are
+    // authenticated.
+    $config = \Drupal::config('og.settings');
+    if ($config->get('group_manager_full_access')) {
+      if (!empty($user_id) && $group_entity instanceof EntityOwnerInterface && $group_entity->getOwnerId() == $user_id) {
+        // @TODO do we need to add the user here?
+        return AccessResult::allowed()
+          ->addCacheableDependency($config);
+      }
+    }
+
+    $pre_alter_cache = [];//static::getPermissionsCache($group_entity, $user, TRUE);
+    $post_alter_cache = [];//static::getPermissionsCache($group_entity, $user, FALSE);
+
+    // To reduce the number of SQL queries, we cache the user's permissions
+    // in a static variable.
+    if (!$pre_alter_cache) {
+      $permissions = array();
+
+      // @todo: Getting permissions from OG Roles will be added here.
+
+      static::setPermissionCache($group_entity, $user, TRUE, $permissions);
+    }
+
+    if (!$skip_alter && !isset($post_alter_cache[$operation])) {
+      // Let modules alter the permissions. So we get the original ones, and
+      // pass them along to the implementing modules.
+      // @todo: Check if still needed to do a clone, since the cache is static
+      // we don't want it to be altered.
+      $alterable_permissions = static::getPermissionsCache($group_entity, $user, TRUE);
+      $context = array(
+        'operation' => $operation,
+        'group' => $group_entity,
+        'user' => $user,
+      );
+
+      \Drupal::moduleHandler()->alter('og_user_access', $alterable_permissions, $context);
+
+      static::setPermissionCache($group_entity, $user, FALSE, $alterable_permissions);
+    }
+
+    $altered_permissions = static::getPermissionsCache($group_entity, $user, FALSE);
+
```

Comment:
@damiankloip this was `$altered_permissions = static::getPermissionsCache($group_entity, $user, TRUE);` but I believe this should be `FALSE` -- we are post alter and the `setPermissionCache` above just set with `FALSE`.